tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPaths.java

```diff
@@ -689,4 +693,84 @@ private int removeNonDistinguishing(
         }
         return counter;
     }
+
+    public void testForUndefined() {
+        DtdVisitor visitor =
+                new DtdVisitor() {
+                    final MatchElementAttribute skipAttributeNames =
+                            new MatchElementAttribute()
+                                    .add( // Add, once checking to make sure that these are safe.
+                                            // Pairs of element, attribute
+                                            "", "references", //
+                                            "", "cp", //
+                                            "version", "", //
+                                            // "ruleset", "type", //
```

Comment:
Commented-out code and empty comments are messy to begin with, and messier due to deep indentation and formatting that breaks up some but not all pairs. Ideally the parameters for add() might be moved to a named constant defined outside the method call, less indented, maybe as an array Map.Entry objects for element/attribute pairs...

The semantics of these elements and attributes are still mostly beyond my understanding. I sometimes wonder whether defining constants for such strings, maybe as members of enums, and referencing them as such rather than as string literals could be helpful for understanding the code and maybe for catching typos...